src/test/app/SetTrust_test.cpp

```diff
@@ -17,50 +17,194 @@
 */
 //==============================================================================
 #include <ripple/test/jtx.h>
+#include <ripple/protocol/TxFlags.h>
+#include <ripple/protocol/JsonFields.h>
 
 namespace ripple {
+
 namespace test {
 
 class SetTrust_test : public beast::unit_test::suite
 {
 public:
-    void testFreeTrustlines()
+
+
+    void testFreeTrustlines(bool thirdLineCreatesLE, bool createOnHighAcct)
     {
-        testcase("Allow 2 free trust lines before requiring reserve");
+        testcase(std::string("Allow two free trust lines before requiring reserve; third line ") +
```

Comment:
That's way too verbose, in my opinion. `testcase ("Dynamic reserve for trust lines")` or `testcase ("First two trustlines are free")`.